pkg/model/bootstrapscript.go

```diff
@@ -98,7 +98,16 @@ func (b *BootstrapScript) buildEnvironmentVariables(cluster *kops.Cluster) (map[
 // template file, substituting in specific env vars & cluster spec configuration
 func (b *BootstrapScript) ResourceNodeUp(ig *kops.InstanceGroup, cluster *kops.Cluster) (*fi.ResourceHolder, error) {
 	// Bastions can have AdditionalUserData, but if there isn't any skip this part
-	if ig.IsBastion() && len(ig.Spec.AdditionalUserData) == 0 {
+	// Check if bastion has hooks enabled, in that special case we want to set hooks up
+	var bastionHooks []kops.HookSpec
+	var err error
+	if ig.IsBastion() {
+		bastionHooks, err = b.getRelevantHooks(ig.Spec.Hooks, ig.Spec.Role)
+	} else {
+		//Do nothing as bastionHooks should be empty
+	}
+
+	if ig.IsBastion() && len(ig.Spec.AdditionalUserData) == 0 && !(len(bastionHooks) > 0) {
```

Comment:
Probably easier on that last term to say `.. && len(bastionHooks) == 0`

Reply:
Fixed in https://github.com/kubernetes/kops/pull/5856/commits/682c5426023c70367628ad08f157fec0577de6b3